src/conn/conn_evict.c

```diff
@@ -0,0 +1,380 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include "wt_internal.h"
+
+#define WT_CONFIG_DEBUG(session, fmt, ...)                                 \
+    if (FLD_ISSET(S2C(session)->debug_flags, WT_CONN_DEBUG_CONFIGURATION)) \
+        __wt_verbose_warning(session, WT_VERB_CONFIGURATION, fmt, __VA_ARGS__);
+
+/*
+ * __evict_config_abs_to_pct --
+ *     Evict configuration values can be either a percentage or an absolute size, this function
+ *     converts an absolute size to a percentage.
+ */
+static WT_INLINE int
+__evict_config_abs_to_pct(
+  WT_SESSION_IMPL *session, double *param, const char *param_name, bool shared)
+{
+    WT_CONNECTION_IMPL *conn;
+    double input;
+
+    conn = S2C(session);
+
+    WT_ASSERT(session, param != NULL);
+    input = *param;
+
+    /*
+     * Anything above 100 is an absolute value; convert it to percentage.
+     */
+    if (input > 100.0) {
+        /*
+         * In a shared cache configuration the cache size changes regularly. Therefore, we require a
+         * percentage setting and do not allow an absolute size setting.
+         */
+        if (shared)
+            WT_RET_MSG(session, EINVAL,
+              "Shared cache configuration requires a percentage value for %s", param_name);
+        /* An absolute value can't exceed the cache size. */
+        if (input > conn->cache_size)
+            WT_RET_MSG(session, EINVAL, "%s should not exceed cache size", param_name);
+
+        *param = (input * 100.0) / (conn->cache_size);
+    }
+
+    return (0);
+}
+
+/*
+ * __validate_evict_config --
```

Comment:
I realize it was poorly named originally and possibly not fully described. Can this be fixed to follow the convention to be something like `__evict_validate_config` or `__evict_config_validate`? But in consideration of that, the function does both the parsing and validation of those values. So the comment is  a bit misleading.